core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -507,9 +507,14 @@ class RequestChannel(val queueSize: Int,
   def sendShutdownRequest(): Unit = requestQueue.put(ShutdownRequest)
 
   def sendCallbackRequest(request: CallbackRequest): Unit = {
-    callbackQueue.put(request)
-    if (!requestQueue.offer(RequestChannel.WakeupRequest))
-      trace("Wakeup request could not be added to queue. This means queue is full, so we will still process callback.")
+    try {
+      callbackQueue.put(request)
+      if (!requestQueue.offer(RequestChannel.WakeupRequest))
+        trace("Wakeup request could not be added to queue. This means queue is full, so we will still process callback.")
+    } catch {
+      case e: Throwable =>
+        warn("There was an exception when attempting to send the callback request", e)
```

Comment:
What happens to the request inside the `CallbackRequest` object? Do we need to send a response?

Reply:
Hmmm -- I guess we would not send a response. Do we send a response normally though if we shutdown during handling a request? Or do we just rely on it to time out?